### PR TITLE
Added decimal_to_binary_recursion.py

### PR DESCRIPTION
**Pull request by cybov**
### **Describe your change:**

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [ ] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: conversions/decimal_to_binary_recursion.py [hunk 1]
```diff
@@ -1,25 +1,23 @@
-""" Convert decimal values to binary using recursion method """
-
-
-def bin_recursive(decimal: int) -> str:
+def binary_recursive(decimal: int) -> str:
     """
-    The funtion takes in a positive integer value
+    This takes in a positive integer value
     and returns its binary equivalent.
```
Comment (cclauss) [inline conversions/decimal_to_binary_recursion.py:4]:
```suggestion
    Take in a positive integer and return its binary equivalent.
```
I tend to use an imperative voice — Just tell the computer what it should do.

File: conversions/decimal_to_binary_recursion.py [hunk 2]
```diff
@@ -1,25 +1,23 @@
...
     Traceback (most recent call last):
     ...
     TypeError: unsupported operand type(s) for //: 'str' and 'int'
     """
     # Initialize exit base of the recursion function
     if decimal == 1 or decimal == 0:
         return str(decimal)
-    return bin_recursive(decimal // 2) + str(decimal % 2)
+    result = binary_recursive(decimal // 2) + str(decimal % 2)
+    return str(result)
```
Comment (cclauss) [inline conversions/decimal_to_binary_recursion.py:20]: I am not a fan of this change.  Creating a variable that has a lifetime of just two lines is not helpful unless it shortens a very long line or the variable name documents something that is not obvious.  `result` in this context does neither.  `binary_recursive()` is documented as returning a str so it is OK to add a str to it.

**Review by cclauss — CHANGES_REQUESTED** [review]

File: conversions/decimal_to_binary_recursion.py (hunk 1 above)
Comment (cybov) [reply conversions/decimal_to_binary_recursion.py]: ok

**Review by cclauss — APPROVED** [review]
Nice!  Please see the modifications that I made.

**Comment by cybov** [comment]
Nice modifications. How do i add the Hacktoberfest label sir?

**Pull request merged**